File: Javascript/Algorithms/Sorting/radixSort.js
const countingSort = (arr, size, place) => {
  
    let output = new Array(size + 1).fill(0);
    let max = Math.max(...arr);
    
    let freq = new Array(max + 1).fill(0);
    
    // Calculate count of elements
    for (let i = 0; i < size; i++){
        const num = Math.floor(arr[i] / place) % 10;
        freq[num]++;
    }
    
    // Calculate cummulative count
    for (let i = 1; i < 10; i++){
        freq[i] += freq[i - 1];
    }
  
    // Place the elements in sorted order
    for (let i = size - 1; i >= 0; i--) {
        const num = Math.floor(arr[i] / place) % 10;
        output[freq[num] - 1] = arr[i];
        freq[num]--;
    }
    
    //Copy the output array
    for (let i = 0; i < size; i++){
      arr[i] = output[i];
    }
  }
  
  const radixSort = (arr, size = arr.length) => {
    //Get the max element
    let max = Math.max(...arr);
    
    //Sort the array using counting sort
    for(let i = 1; parseInt(max / i) > 0; i *= 10){
      countingSort(arr, size, i);
    }
  }

File: C++/Algorithms/Sorting/radixsort.cpp
#include <iostream>
using namespace std;
 
int getMax(int arr[], int n)
{
    int mx = arr[0];
    for (int i = 1; i < n; i++)
        if (arr[i] > mx)
            mx = arr[i];
    return mx;
}
 

void countSort(int arr[], int n, int exp)
{
    int output[n]; 
    int i, count[10] = { 0 };
 
    for (i = 0; i < n; i++)
        count[(arr[i] / exp) % 10]++;
 
    for (i = 1; i < 10; i++)
        count[i] += count[i - 1];
 
    for (i = n - 1; i >= 0; i--) {
        output[count[(arr[i] / exp) % 10] - 1] = arr[i];
        count[(arr[i] / exp) % 10]--;
    }
 
    for (i = 0; i < n; i++)
        arr[i] = output[i];
}

void radixsort(int arr[], int n)
{
    int m = getMax(arr, n);
 
    for (int exp = 1; m / exp > 0; exp *= 10)
        countSort(arr, n, exp);
}
 
void print(int arr[], int n)
{
    for (int i = 0; i < n; i++)
        cout << arr[i] << " ";
}
 
int main()
{
    int n;
    cin>>n;

    int arr[n];

    for(int i=0;i<n;i++){
        cin>>arr[i];
    }
     
    radixsort(arr, n);
    print(arr, n);
    return 0;
}